src/ORM/EagerLoadedList.php

```diff
@@ -0,0 +1,1003 @@
+     */
+    public function dataClass(): string
+    {
+        return $this->dataClass;
+    }
+
+    public function dbObject($fieldName): ?DBField
+    {
+        return singleton($this->dataClass)->dbObject($fieldName);
+    }
+
+    public function getIDList(): array
+    {
+        $ids = $this->column('ID');
+        return array_combine($ids, $ids);
+    }
+
+    /**
+     * Sets the ComponentSet to be the given ID list
+     * @throws BadMethodCallException
+     */
+    public function setByIDList($idList): void
+    {
+        throw new BadMethodCallException("Can't set the ComponentSet on an EagerLoadedList");
+    }
+
+    /**
+     * Returns a copy of this list with the relationship linked to the given foreign ID
+     * @throws BadMethodCallException
+     */
+    public function forForeignID($id): void
+    {
+        throw new BadMethodCallException("Can't change the foreign ID for an EagerLoadedList");
+    }
+
+    public function getIterator(): Traversable
+    {
+        $limitedRows = $this->getFinalisedRows();
+        foreach ($limitedRows as $row) {
+            yield $this->createDataObject($row);
+        }
+    }
+
+    /**
+     * Get the raw data rows for the records in this list.
+     * Doesn't include nested eagerloaded data.
+     */
+    public function getRows(): array
+    {
+        return array_values($this->rows);
+    }
+
+    public function toArray(): array
+    {
+        $result = [];
+        foreach ($this as $item) {
+            $result[] = $item;
+        }
+        return $result;
+    }
+
+    public function toNestedArray(): array
+    {
+        $result = [];
+        foreach ($this as $item) {
+            $result[] = $item->toMap();
+        }
+        return $result;
+    }
+
+    /**
+     * Add a single row of database data.
+     *
+     * @throws InvalidArgumentException if there is no ID in $row
+     */
+    public function addRow(array $row): static
+    {
+        if (!array_key_exists('ID', $row) || $row['ID'] === null || $row['ID'] === '' || is_array($row['ID'])) {
+            throw new InvalidArgumentException('$row must have a valid ID');
+        }
+        $this->rows[$row['ID']] = $row;
+        return $this;
+    }
+
+    /**
+     * Add multiple rows of database data.
+     *
+     * @throws InvalidArgumentException if any row is missing an ID
+     */
+    public function addRows(array $rows): static
+    {
+        foreach ($rows as $row) {
+            $this->addRow($row);
+        }
+        return $this;
+    }
+
+    /**
+     * Not implemented - use addRow instead.
+     */
+    public function add($item)
+    {
+        throw new BadMethodCallException('Cannot add a DataObject record to EagerLoadedList. Use addRow() to add database rows.');
+    }
+
+    /**
+     * Removes a record from the list. Note that the record will not be removed from the
+     * database - this list is read-only.
+     */
+    public function remove($obj): static
```

Comment:
```suggestion
    public function remove($item): static
```

Give it the same param name as DataList and add()

Reply:
Done